sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/collect.scala

```diff
@@ -330,8 +330,14 @@ case class ListAgg(
 
   override def isDistinctSupported: Boolean = true
 
-  override def withOrderingWithinGroup(orderingWithinGroup: Seq[SortOrder]): AggregateFunction =
+  override def withOrderingWithinGroup(
+      orderingWithinGroup: Seq[SortOrder], u: UnresolvedFunction): AggregateFunction = {
+    if (u.isDistinct && !isOrderCompatible(orderingWithinGroup)) {
```

Comment:
does it mean all functions need to validate it by themselves?

Reply:
I think the special behavior should be validated by themselves if we can't find a common way.

Reply:
I think the special behavior should be validated by themselves if we can't find a common way.